lib/coordinatorsharding.go

```diff
@@ -386,8 +387,9 @@ func (crd *Coordinator) PreprocessSharding(requests []*netstring.Netstring) (boo
 						}
 						evt := cal.NewCalEvent(EvtTypeSharding, EvtNameNoShardKey, cal.TransOK, "")
 						evt.AddDataInt("sql", int64(uint32(crd.sqlhash)))
+						evt.AddDataStr("shard_key", GetConfig().ShardKeyName)
 						evt.Completed()
-						ns := netstring.NewNetstringFrom(common.RcError, []byte(ErrNoShardKey.Error()))
+						ns := netstring.NewNetstringFrom(common.RcError, []byte(fmt.Sprintf(ErrNoShardKey.Error(), GetConfig().ShardKeyName)))
 						crd.respond(ns.Serialized)
 						return false /*don't hangup*/, ErrNoShardKey
```

Comment:
We must do a Sprintf to the error before return. Otherwise, 
https://github.com/paypal/hera/blob/master/lib/coordinator.go#L410 would show up as `373: no shard key '%s'`

Reply:
agree, alternatively since shardkey is not changing during runtime, it may be feasible to avoid using fmt.Sprintf with %s.

Reply:
removed place holders in constant error string